capstone-rs/src/instruction.rs

```diff
@@ -35,8 +35,19 @@ pub type RegIdInt = u16;
 
 /// Represents an register id, which is architecture-specific.
 #[derive(Copy, Clone, Debug, Hash, PartialEq, Eq, PartialOrd, Ord)]
+#[repr(transparent)]
 pub struct RegId(pub RegIdInt);
 
+impl core::convert::From<u32> for RegId {
+    fn from(v: u32) -> RegId {
+        if v <= core::u16::MAX as u32 {
```

Comment:
We should use `TryFrom` instead. Also, use `RegId` instead of `u16` directly.

```rust
RegId(x.try_into().ok().unwrap_or(0))
```

https://doc.rust-lang.org/std/primitive.u16.html#impl-TryFrom%3Cu32%3E